libraries/standard/http/src/http_client.c

```diff
@@ -1192,6 +1210,72 @@ static HTTPStatus_t addHeader( HTTPRequestHeaders_t * pRequestHeaders,
 
 /*-----------------------------------------------------------*/
 
+static HTTPStatus_t addRangeHeader( HTTPRequestHeaders_t * pRequestHeaders,
+                                    int32_t rangeStartOrlastNbytes,
+                                    int32_t rangeEnd )
+{
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+
+    /* This buffer uses a char type instead of the general purpose uint8_t because
+    * the range value expected to be written is within the ASCII character set. */
+    char rangeValueBuffer[ HTTP_MAX_RANGE_REQUEST_VALUE_LEN ] = { '\0' };
+    size_t rangeValueLength = 0u;
+
+    /* Generate the value data for the Range Request header.*/
+
+    /* Write the range value prefix in the buffer. */
+    ( void ) memcpy( rangeValueBuffer,
+                     HTTP_RANGE_REQUEST_HEADER_VALUE_PREFIX,
+                     HTTP_RANGE_REQUEST_HEADER_VALUE_PREFIX_LEN );
+    rangeValueLength += HTTP_RANGE_REQUEST_HEADER_VALUE_PREFIX_LEN;
+
+    /* Write the range start value in the buffer. */
+    rangeValueLength += convertInt32ToAscii( rangeStartOrlastNbytes,
+                                             rangeValueBuffer + rangeValueLength,
+                                             sizeof( rangeValueBuffer ) - rangeValueLength );
+
+    /* Add remaining value data depending on the range specification type. */
+
+    /* Add rangeEnd value if request is for [rangeStart, rangeEnd] byte range */
+    if( rangeEnd != HTTP_RANGE_REQUEST_END_OF_FILE )
+    {
+        /* Write the "-" character to the buffer.*/
+        ( void ) memcpy( rangeValueBuffer + rangeValueLength,
+                         DASH_CHARACTER,
+                         DASH_CHARACTER_LEN );
+        rangeValueLength += DASH_CHARACTER_LEN;
+
+        /* Write the rangeEnd value of the request range to the buffer .*/
+        rangeValueLength += convertInt32ToAscii( rangeEnd,
+                                                 rangeValueBuffer + rangeValueLength,
+                                                 sizeof( rangeValueBuffer ) - rangeValueLength );
+    }
+    /* Case when request is for bytes in the range [rangeStart, EoF). */
+    else if( rangeStartOrlastNbytes >= 0 )
+    {
+        /* Write the "-" character to the buffer.*/
+        ( void ) memcpy( rangeValueBuffer + rangeValueLength,
```

Comment:
So I realized here that instead of using memcpy we should have just assigned the single dash character to the buffer lol

```
*( rangeValueBuffer + rangeValueLength ) = DASH_CHARACTER;
```